Neos.Flow/Tests/Unit/Http/Component/SessionRequestComponentTest.php

```diff
@@ -0,0 +1,216 @@
+            ['sessionCookieSettings' => ['lifetime' => 123], 'expectedNewCookieValue' => 'session_cookie_name=session-id; Max-Age=123; Path=/; HttpOnly'],
+            ['sessionCookieSettings' => ['path' => '/some/path'], 'expectedNewCookieValue' => 'session_cookie_name=session-id; Path=/some/path; HttpOnly'],
+            ['sessionCookieSettings' => ['secure' => true], 'expectedNewCookieValue' => 'session_cookie_name=session-id; Path=/; Secure; HttpOnly'],
+            ['sessionCookieSettings' => ['httponly' => false], 'expectedNewCookieValue' => 'session_cookie_name=session-id; Path=/'],
+            ['sessionCookieSettings' => ['domain' => 'neos.io'], 'expectedNewCookieValue' => 'session_cookie_name=session-id; Domain=neos.io; Path=/; HttpOnly'],
+            ['sessionCookieSettings' => ['samesite' => 'none'], 'expectedNewCookieValue' => 'session_cookie_name=session-id; Path=/; Secure; HttpOnly; SameSite=none'],
+            ['sessionCookieSettings' => ['samesite' => 'strict'], 'expectedNewCookieValue' => 'session_cookie_name=session-id; Path=/; HttpOnly; SameSite=strict'],
+            ['sessionCookieSettings' => ['samesite' => 'lax'], 'expectedNewCookieValue' => 'session_cookie_name=session-id; Path=/; HttpOnly; SameSite=lax'],
+        ];
+    }
+
+    /**
+     * @test
+     * @dataProvider sessionCookieSettingsProvider
+     */
+    public function newSessionCookiesTakeSessionCookieSettingsIntoAccount(array $sessionCookieSettings, string $expectedCookie): void
+    {
+        $this->mockHttpRequest->method('getCookieParams')->willReturn(['session_cookie_name' => 'session-id']);
+
+        $this->inject($this->sessionRequestComponent, 'sessionSettings', [
+            'name' => 'session_cookie_name',
+            'cookie' => array_merge($this->defaultSessionCookieSettings, $sessionCookieSettings),
+        ]);
+
+        $this->mockSessionManager->expects($this->once())->method('initializeCurrentSessionFromCookie')->willReturnCallback(static function (Cookie $cookie) use ($expectedCookie) {
+            self::assertSame($expectedCookie, (string)$cookie);
+        });
+
+        $this->sessionRequestComponent->handle($this->mockComponentContext);
+    }
+
+    public function cookieValueDataProvider(): array
+    {
+        return [
+            ['sessionCookieValue' => 123, 'expectedNewCookieValue' => '123'],
+            ['sessionCookieValue' => '', 'expectedNewCookieValue' => ''],
+            ['sessionCookieValue' => 'some String', 'expectedNewCookieValue' => 'some String'],
+            ['sessionCookieValue' => '"leading quote', 'expectedNewCookieValue' => 'leading quote'],
+            ['sessionCookieValue' => 'trailing quote"', 'expectedNewCookieValue' => 'trailing quote'],
+            ['sessionCookieValue' => '"quotes"', 'expectedNewCookieValue' => 'quotes'],
+            ['sessionCookieValue' => '""double quotes"', 'expectedNewCookieValue' => 'double quotes'],
```

Comment:
Tests ❤️

See comment above, but I guess this test states that you think we should just do it like this?

Reply:
Yes, for now at least